src/disk_io_thread_pool.cpp

```diff
@@ -103,6 +101,8 @@ namespace libtorrent {
 
 	bool disk_io_thread_pool::try_thread_exit(std::thread::id id)
 	{
+		if (m_abort) return true;
```

Comment:
I'm not sure about this one. the other case where we return true, we also remove the thread from `m_threads`. I should try remove this

Reply:
there would need to be some special case below though, to remove threads when aborting

Reply:
the important part is that when we exit via this mechanism, we always make sure the job queue is empty first

Reply:
it's also accessing `m_abort` without holding `m_mutex`...